containers/helm.extern.d.ts

```diff
@@ -0,0 +1,282 @@
+  readonly addOwnerReference: (owner: OwnerReference) => void;
+  /** 
+  @returns a value of a label or undefined */
+  readonly getLabel: (key: string) => (string) | undefined;
+  /** The name of the API object.
+  If a name is specified in `metadata.name` this will be the name returned.
+  Otherwise, a name will be generated by calling
+  `Chart.of(this).generatedObjectName(this)`, which by default uses the
+  construct path to generate a DNS-compatible name for the resource. */
+  readonly name?: (string) | undefined;
+  /** The object's namespace. */
+  readonly namespace?: (string) | undefined;
+  /** Synthesizes a k8s ObjectMeta for this metadata set. */
+  readonly toJson: () => any;
+}
+export class ApiObject extends Construct {
+  /** Create a dependency between this ApiObject and other constructs.
+  These can be other ApiObjects, Charts, or custom. */
+  readonly addDependency: (dependencies?: ((readonly (IConstruct)[])) | undefined) => void;
+  /** Applies a set of RFC-6902 JSON-Patch operations to the manifest synthesized for this API object.
+    kubePod.addJsonPatch(JsonPatch.replace('/spec/enableServiceLinks', true)); */
+  readonly addJsonPatch: (ops?: ((readonly (JsonPatch)[])) | undefined) => void;
+  /** The group portion of the API version (e.g. `authorization.k8s.io`). */
+  readonly apiGroup: string;
+  /** The object's API version (e.g. `authorization.k8s.io/v1`). */
+  readonly apiVersion: string;
+  /** The chart in which this object is defined. */
+  readonly chart: Chart;
+  /** The object kind. */
+  readonly kind: string;
+  /** Metadata associated with this API object. */
+  readonly metadata: ApiObjectMetadataDefinition;
+  /** The name of the API object.
+  If a name is specified in `metadata.name` this will be the name returned.
+  Otherwise, a name will be generated by calling
+  `Chart.of(this).generatedObjectName(this)`, which by default uses the
+  construct path to generate a DNS-compatible name for the resource. */
+  readonly name: string;
+  /** Renders the object to Kubernetes JSON.
+  To disable sorting of dictionary keys in output object set the
+  `CDK8S_DISABLE_SORT` environment variable to any non-empty value. */
+  readonly toJson: () => any;
+}
+export class Chart extends Construct {
+  /** Create a dependency between this Chart and other constructs.
+  These can be other ApiObjects, Charts, or custom. */
+  readonly addDependency: (dependencies?: ((readonly (IConstruct)[])) | undefined) => void;
+  /** Returns all the included API objects. */
+  readonly apiObjects: (readonly (ApiObject)[]);
+  /** Generates a app-unique name for an object given it's construct node path.
+  Different resource types may have different constraints on names
+  (`metadata.name`). The previous version of the name generator was
+  compatible with DNS_SUBDOMAIN but not with DNS_LABEL.
+  
+  For example, `Deployment` names must comply with DNS_SUBDOMAIN while
+  `Service` names must comply with DNS_LABEL.
+  
```

Comment:
@MarkMcCulloh this definitely feels off.

Reply:
What do you mean?

Reply:
This generated .d.ts file.

(related to our discussion earlier about source of truth)